src/mscorlib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -208,7 +208,7 @@ internal static bool TryParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> form
                     return (true);
                 }
             }
-            result.SetFailure(ParseFailureKind.Format, nameof(SR.Format_BadDateTime), null);
+            result.SetFailure(ParseFailureKind.FormatWithOriginalDateTime, nameof(SR.Format_BadDateTime), null);
```

Comment:
Sometimes the last parameter is null sometimes STring.Empty. I think the effect is the same. If you want you could add an overload of SetFailure that has only 2 parameters and eliminate all these.